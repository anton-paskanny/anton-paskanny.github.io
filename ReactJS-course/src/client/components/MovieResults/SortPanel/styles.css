.sort-panel {
    display: flex;
    justify-content: space-between;
    padding: 10px 20%;
    background-color: var(--third-color);
}

.sort-panel__filters {
    display: flex;
}

.sort-panel__inner-wrapper {
    display: flex;
}

.sort-panel__filter-item {
    margin-left: 10px;
}

.sort-panel__filter-btn {
    padding: 0;
    border: none;
    background-color: transparent;
}

.sort-panel__filter-btn--active {
    color: var(--primary-color);
}