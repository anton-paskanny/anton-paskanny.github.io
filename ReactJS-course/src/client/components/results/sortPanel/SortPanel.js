import React from 'react';

const SortPanel = () => (
    <div>
        Sort Panel
    </div>
)

export default SortPanel;