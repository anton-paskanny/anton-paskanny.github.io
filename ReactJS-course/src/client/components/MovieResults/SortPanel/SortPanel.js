import React from 'react';

import { SORT_BY_CONFIG } from '../../../utils'

import styles from './styles.css';


const SortPanel = props => {

    const renderSortByItems = () => {
        return SORT_BY_CONFIG.map((item) => (
                <li key={item.name} className='sort-panel__filter-item'>
                    <button className={'sort-panel__filter-btn' + (item.fieldToSortBy === props.sortType ? ' sort-panel__filter-btn--active': '')}>{item.name}</button>
                </li>
            )
        );
    }

    const handleSortByChange = (e) => {
        if (e.target.classList.contains('sort-panel__filter-btn--active')) {
            return;
        }

        const sortItem = SORT_BY_CONFIG.find(item => item.name === e.target.textContent);

        sortItem && sortItem.fieldToSortBy && props.toggleSort(sortItem.fieldToSortBy);
    }

    const renderContent = () => (
        <div className="sort-panel__inner-wrapper">
            <p>Sort by</p>
            <ul className="sort-panel__filters" onClick={handleSortByChange}>
                {renderSortByItems()}
            </ul>
        </div>
    );

    const renderCounterContent = () => {
        if (props.selectedMovie) {
            const genres = props.selectedMovie.genres.map(genre => genre).join(', ');

            return `Films by ${genres} ${props.selectedMovie.genres.length > 1 ? 'genres' : 'genre'}`;
        }

        return `${props.movies.length} ${props.movies.length > 1 ? 'movies' : 'movie'} found`;
    };

    return (
        <div className="sort-panel">
            <p className="sort-panel__results-counter">
                {renderCounterContent()}
            </p>
            {!props.selectedMovie && renderContent()}
        </div>
    )
}

export default SortPanel;