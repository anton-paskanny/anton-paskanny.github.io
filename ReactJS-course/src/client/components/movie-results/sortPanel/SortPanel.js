import React from 'react';

import styles from './styles.css';

const SortPanel = () => (
    <div className="sort-panel">
        <p className="sort-panel__results-counter">
            7 movies found
        </p>
        <div className="sort-panel__inner-wrapper">
            <p>Sort by</p>
            <ul className="sort-panel__filters">
                <li className="sort-panel__filter-item sort-panel__filter-item--active">release date</li>
                <li className="sort-panel__filter-item">rating</li>
            </ul>
        </div>
    </div>
)

export default SortPanel;