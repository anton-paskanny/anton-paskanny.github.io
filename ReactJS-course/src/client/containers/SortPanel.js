import { connect } from 'react-redux';
import { toggleSort } from '../actions/sortPanel';

import SortPanel from '../components/MovieResults/SortPanel/SortPanel';

export const mapStateToProps = state => ({
  movies: state.getIn(['movies', 'data']),
  selectedMovie: state.getIn(['selectedMovie', 'data']),
  sortType: state.getIn(['sort', 'type']),
});

export const mapDispatchToProps = {
  toggleSort,
};

export default connect(mapStateToProps, mapDispatchToProps)(SortPanel);
