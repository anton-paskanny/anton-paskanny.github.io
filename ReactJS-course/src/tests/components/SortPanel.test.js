import React from 'react';
import { shallow } from 'enzyme';

import SortPanel from '../../client/components/MovieResults/SortPanel/SortPanel';

import { movieItem, movies } from '../mocks/dataMock';

let component, props, sortByConfig;

describe('SortPanel', () => {
    beforeEach(() => {
        sortByConfig = [
            {
                name: 'release date',
                fieldToSortBy: 'release_date',
                active: true
            },
            {
                name: 'rating',
                fieldToSortBy: 'vote_average',
                active: false
            }
        ];
        props = { sortByConfig, movies };
        component = shallow(<SortPanel {...props} />);
    });
    it('render component correctly', () => {
        expect(component).toMatchSnapshot();
    });
    it('should contain list with sort items', () => {
        expect(component.find('.sort-panel__filter-item')).toHaveLength(sortByConfig.length);
    });
    it('should contain several genres in render output', () => {
        component.setProps({
            selectedMovie: movieItem
        });
        expect(component.find('.sort-panel__results-counter').text()).toMatch(/genres/);
    });
    it('should contain one genre in render output', () => {
        movieItem.genres = ['Adventure'];
        component.setProps({
            selectedMovie: movieItem
        });
        expect(component.find('.sort-panel__results-counter').text()).toMatch(/genre/);
    });
    it('should contain one movie in render output', () => {
        movieItem.genres = ['Adventure'];
        component.setProps({
            movies: [ movieItem ]
        });
        expect(component).toMatchSnapshot();
        expect(component.find('.sort-panel__results-counter').text()).toMatch(/movie/);
    });
    it('should not rendere anything when no movies were received', () => {
        component.setProps({
            movies: []
        });
        expect(component).toMatchSnapshot();
        expect(component.find('.sort-panel').length).toBe(0);
    });
});



