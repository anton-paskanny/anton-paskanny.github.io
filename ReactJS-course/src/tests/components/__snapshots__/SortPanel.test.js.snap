// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortPanel render component correctly 1`] = `
<div
  className="sort-panel"
>
  <p
    className="sort-panel__results-counter"
  >
    4 movies found
  </p>
  <div
    className="sort-panel__inner-wrapper"
  >
    <p>
      Sort by
    </p>
    <ul
      className="sort-panel__filters"
    >
      <li
        className="sort-panel__filter-item"
        key="release date"
      >
        <button
          className="sort-panel__filter-btn sort-panel__filter-btn--active"
        >
          release date
        </button>
      </li>
      <li
        className="sort-panel__filter-item"
        key="rating"
      >
        <button
          className="sort-panel__filter-btn"
        >
          rating
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`SortPanel should contain one movie in render output 1`] = `
<div
  className="sort-panel"
>
  <p
    className="sort-panel__results-counter"
  >
    1 movie found
  </p>
  <div
    className="sort-panel__inner-wrapper"
  >
    <p>
      Sort by
    </p>
    <ul
      className="sort-panel__filters"
    >
      <li
        className="sort-panel__filter-item"
        key="release date"
      >
        <button
          className="sort-panel__filter-btn sort-panel__filter-btn--active"
        >
          release date
        </button>
      </li>
      <li
        className="sort-panel__filter-item"
        key="rating"
      >
        <button
          className="sort-panel__filter-btn"
        >
          rating
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`SortPanel should not rendere anything when no movies were received 1`] = `""`;
